Networking/Networking/Model/Product/ProductDownloadDragAndDrop.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import CoreServices
+
+/// A wrapper around `ProductDownload`, to make it compatible in using as Drag and Drop data source in a table view. Represents a `ProductDownload` entity.
+/// To make a data draggable and droppable, on an Table/Collection view,
+/// the data source object needs to confirm to `NSItemProviderReading`
+/// and `NSItemProviderWriting` protocol and these 2 protocol again confirms to `NSObjectProtocol`
+/// So the top layer object needs to be a subclass of `NSObject`
+/// And since the original `ProductDownload `is a struct, so we need a new class for this purpose.
+///
+public final class ProductDownloadDragAndDrop: NSObject, Codable {
+    public let downloadableFile: ProductDownload
+
+    /// initializer.
+    ///
+    public init(downloadableFile: ProductDownload) {
+        self.downloadableFile = downloadableFile
+    }
```

Comment:
It seems that here is missing `super.init()`
```suggestion
        self.downloadableFile = downloadableFile
        super.init()
    }
```

Reply:
Updated